simulator/src/json.rs

```diff
@@ -0,0 +1,220 @@
+use chrono::{DateTime, Utc};
+use rand::{Rng, SeedableRng};
+use rand_distr::{Distribution, Exp, Normal};
+use serde::Deserialize;
+use std::ops::RangeInclusive;
+use supermusr_common::{Channel, DigitizerId, FrameNumber, Time};
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "kebab-case", untagged)]
+pub(crate) enum Expression {
+    Fixed(f64),
+    FrameTransform(Transformation<f64>),
+}
+
+impl Expression {
+    fn value(&self, frame_index: usize) -> f64 {
+        match self {
+            Expression::Fixed(v) => *v,
+            Expression::FrameTransform(frame_function) => {
+                frame_function.transform(frame_index as f64)
+            }
+        }
+    }
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "kebab-case", untagged)]
```

Comment:
I'm quite against `untagged` as it adds a layer of obfuscation as to what you are defined, prefer `tag = "type"` as you have used elsewhere.

Reply:
I used `untagged`, as fields of type `Expression`, will automatically coerce a float to Expression::Fixed(float). This allows the user to write either a literal or a `Transformation` (a linear function depending on the frame number). As most values are likely to be literals this avoids the user having to clutter up the file with things like `{"type": "Fixed", "value": 35.0}`.
The same applies to fields of type `RandomDistribution` and `Frames`, though to a lesser extent, so perhaps it makes sense to only use `untagged` for `Expression`?

Reply:
I don't think the benefits of such syntactic niceties are worth the validation/type safety that explicit specifying what you want bring. I would rather not use `untagged` anywhere.